pkg/virt-controller/watch/vm.go

```diff
@@ -2900,38 +2854,53 @@ func (c *VMController) trimDoneVolumeRequests(vm *virtv1.VirtualMachine) {
 }
 
 // addRestartRequiredIfNeeded adds the restartRequired condition to the VM if any non-live-updatable field was changed
-func (c *VMController) addRestartRequiredIfNeeded(lastSeenVMSpec *virtv1.VirtualMachineSpec, vm *virtv1.VirtualMachine) (*virtv1.VirtualMachine, bool) {
+func (c *VMController) addRestartRequiredIfNeeded(lastSeenVMSpec *virtv1.VirtualMachineSpec, vm *virtv1.VirtualMachine) bool {
 	if lastSeenVMSpec == nil {
-		return vm, false
+		return false
+	}
+
+	// Expand any instance types and preferences associated with lastSeenVMSpec or the current VM before working out if things are live-updatable
+	currentVM := vm.DeepCopy()
+	if err := c.instancetypeMethods.ApplyToVM(currentVM); err != nil {
+		return false
+	}
+	lastSeenVM := &virtv1.VirtualMachine{
```

Comment:
Is there a way to passs the `lastSeenVM` to this func instead of the `lastSeenVMSpec`?

Reply:
I can move this to just before the caller but I'm not sure it really gains us anything?